python_files/generate_review_figures.py

```diff
@@ -0,0 +1,625 @@
+                                               'distance_to__NK', 'distance_to__Other', 'distance_to__Structural', 'distance_to__T']]
+
+]
+per_img_stats=[
+    ['fiber', fiber_df],
+    ['fiber', fiber_tile_df],
+    ['mixing_score', mixing_df],
+    ['kmeans_cluster', kmeans_img_proportions],
+    ['kmeans_cluster', kmeans_compartment_proportions],
+    ['pixie_ecm', pixie_ecm],
+    ['ecm_fraction', ecm_frac],
+    ['ecm_cluster', ecm_clusters],
+    ['cell_interactions', cell_interactions]
+]
+
+features = SpaceCat(adata, image_key='fov', seg_label_key='label', cell_area_key='area',
+                    cluster_key=['cell_cluster', 'cell_cluster_broad'],
+                    compartment_key='compartment', compartment_area_key='compartment_area')
+# Generate features and save anndata
+adata_processed = features.run_spacecat(functional_feature_level='cell_cluster', diversity_feature_level='cell_cluster', pixel_radius=50,
+                                        specified_ratios_cluster_key='cell_cluster', specified_ratios=ratio_pairings,
+                                        per_cell_stats=per_cell_stats, per_img_stats=per_img_stats)
+
+adata_processed.write_h5ad(os.path.join(data_dir, 'adata', 'adata_processed.h5ad'))
+
+# plot prediction comparison
+preds = pd.read_csv('/Users/csowers/Documents/TONIC_reviews/SpaceCat/prediction_model/patient_outcomes/all_timepoints_results_MIBI.csv')
+preds = preds[['auc_primary_list', 'auc_baseline_list', 'auc_post_induction_list', 'auc_on_nivo_list']]
+preds = preds.rename(columns={'auc_primary_list': 'Primary', 'auc_baseline_list': 'Baseline',
+                              'auc_post_induction_list': 'Pre nivo', 'auc_on_nivo_list': 'On nivo'})
+preds = preds.melt()
+preds['Analysis'] = 'excluding'
+
+adj_preds = pd.read_csv('/Users/csowers/Documents/TONIC_reviews/NT_cell_interactions/prediction_model/patient_outcomes/all_timepoints_results_MIBI.csv')
+adj_preds = adj_preds[['auc_primary_list', 'auc_baseline_list', 'auc_post_induction_list', 'auc_on_nivo_list']]
+adj_preds = adj_preds.rename(columns={'auc_primary_list': 'Primary', 'auc_baseline_list': 'Baseline',
+                                      'auc_post_induction_list': 'Pre nivo', 'auc_on_nivo_list': 'On nivo'})
+adj_preds = adj_preds.melt()
+adj_preds['Analysis'] = 'including'
+all_preds = pd.concat([preds, adj_preds])
+
+fig, ax = plt.subplots()
+sns.boxplot(data=all_preds, x='variable', y='value', ax=ax, width=0.6, hue='Analysis',
+            palette=sns.color_palette(["blue", "yellow"]), boxprops=dict(alpha=.3))
+plt.xticks(rotation=30)
+plt.title('TONIC prediction including and excluding interaction features')
+plt.ylabel('AUC')
+plt.xlabel('')
+plt.legend(loc='lower right').set_title('')
+plt.savefig(os.path.join(NT_interactions_viz_dir, 'cell_interaction_prediction_comparison.pdf'), bbox_inches='tight', dpi=300)
```

Comment:
Will you use same plot scheme as main figure AUC boxplots?

Reply:
<img width="625" alt="Screenshot 2024-12-16 at 2 51 57 PM" src="https://github.com/user-attachments/assets/f42585a1-4f61-4f1e-a0c9-24e690d11e17" />

Reply:
<img width="626" alt="Screenshot 2024-12-16 at 2 54 05 PM" src="https://github.com/user-attachments/assets/85844ebe-9466-4564-a792-b4e42369d668" />